industrial_extrinsic_cal/src/check_if_points_in_pic.cpp

```diff
@@ -0,0 +1,349 @@
+#include <industrial_extrinsic_cal/targets_yaml_parser.h>
+#include <ros/ros.h>
+#include <ros/node_handle.h>
+#include <geometry_msgs/PoseArray.h>
+#include <eigen_conversions/eigen_msg.h>
+#include "applicable_pose_generator/pose_reachability_filter.h"
+
+#include <geometry_msgs/TransformStamped.h>
+#include <geometry_msgs/Twist.h>
+
+#include <industrial_extrinsic_cal/ros_transform_interface.h>
+#include <industrial_extrinsic_cal/check_if_points_in_pic.h>
+
+#include <industrial_extrinsic_cal/camera_yaml_parser.h>
+
+using std::vector;
+using std::string;
+using industrial_extrinsic_cal::Target;
+using YAML::Node;
+
+// local function prototypes
+geometry_msgs::PoseArray pose_filters(geometry_msgs::PoseArray msg2, tf::StampedTransform tf_transform, int image_width, int image_height, EigenSTL::vector_Affine3d AllcameraPoses,Eigen::Vector3d corner_points[4],double fx, double fy, double cx, double cy );
+void imagePoint(Eigen::Vector3d TargetPoint, double fx, double fy, double cx, double cy, double &u, double &v,Eigen::Affine3d cameraPose);
+geometry_msgs::PoseArray create_all_poses(double poseHeight, double spacing_in_z, double angleOfCone, int numberOfStopsForPhotos, Eigen::Vector2d center_point_of_target );
+
+int addingFactorial(int lastAdded);
+Eigen::Vector2d finds_middle_of_target(Eigen::Vector3d corner_points[4], double center_Of_TargetX, double center_Of_TargetY);
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "node");
+  sleep(2);
+  ros::NodeHandle n("~");
+  int image_width =  1080;
+  int image_height = 1080;
+  double fy=529.4;
+  double fx =529.2;
+  int numberOfStopsForPhotos= 8; // controlls amount of stops for each ring
+  double poseHeight = 1.2;
+  double angleOfCone = M_PI/3;
+  double spacing_in_z = .25;
+  double center_Of_TargetX;
+  double center_Of_TargetY;
+  Eigen::Vector2d center_point_of_target;
+  tf::StampedTransform tf_transform;
+  tf::TransformListener tf_listener;
+  double xMax,yMax,xMin,yMin;
+
+  //gets camera parameters
+  ros::NodeHandle pivnh("~");
+  if (!pivnh.getParam("image_width", image_width)){
+    ROS_ERROR("did not set parameter image_width");
+  }
+  if (!pivnh.getParam("image_height", image_height)){
+    ROS_ERROR("did not set parameter image_height");
+  }
+  if (!pivnh.getParam("focal_length", fx)){
+    ROS_ERROR("did not set parameter focal_length");
+  }
+
+  fy = fx;
+
+  double cx = image_width/2;//assumes that lense is ligned up with sensor
+  double cy = image_height/2;//assums that lense is ligned up with sensor
+
+  make_main_smaller initialization;
+  initialization.create_transform_listener(tf_transform, tf_listener);
+
+  //reads in information about target
+  string targetFile = "/home/lawrencelewis/catkin_ws/src/industrial_calibration/industrial_extrinsic_cal/yaml/ical_srv_target.yaml";
+  vector<boost::shared_ptr<Target>>  myYamlDefinedTargets;
+  if(!parseTargets(targetFile , myYamlDefinedTargets))
+{
+    ROS_ERROR("file name bad");
```

Comment:
provide more information, like 
ROS_ERROR("could not parse target file %s", targetFile.c_str());